src/tlv/TlvCodec.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright 2022 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DataReaderLE } from "../util/DataReaderLE";
+import { DataWriterLE } from "../util/DataWriterLE";
+import { BitmapSchema, EnumBits } from "../util/schema/BitmapSchema";
+
+
+/**
+ * TLV element types.
+ * 
+ * @see {@link MatterCoreSpecificationV1_0} § A.7.1
+ */
+export enum TlvType {
+    SignedInt_1OctetValue = 0x00,
+    SignedInt_2OctetValue = 0x01,
+    SignedInt_4OctetValue = 0x02,
+    SignedInt_8OctetValue = 0x03,
+    UnsignedInt_1OctetValue = 0x04,
+    UnsignedInt_2OctetValue = 0x05,
+    UnsignedInt_4OctetValue = 0x06,
+    UnsignedInt_8OctetValue = 0x07,
+    Boolean = 0x08,
+    Float = 0x0A,
+    Double = 0x0B,
+    Utf8String_1OctetLength = 0x0C,
+    Utf8String_2OctetLength = 0x0D,
+    Utf8String_4OctetLength = 0x0E,
+    Utf8String_8OctetLength = 0x0F,
+    OctetString_1OctetLength = 0x10,
+    OctetString_2OctetLength = 0x11,
+    OctetString_4OctetLength = 0x12,
+    OctetString_8OctetLength = 0x13,
+    Null = 0x14,
+    Structure = 0x15,
+    Array = 0x16,
+    List = 0x17,
+    EndOfContainer = 0x18,
+}
+
+/**
+ * TLV element tag control.
+ * 
+ * @see {@link MatterCoreSpecificationV1_0} § A.7.2
+ */
+const enum TagControl {
+    Anonymous = 0,
+    ContextSpecific = 1,
+    CommonProfile2Bytes = 2,
+    CommonProfile4Bytes = 3,
+    ImplicitProfile2Bytes = 4,
+    ImplicitProfile4Bytes = 5,
+    FullyQualified6Bytes = 6,
+    FullyQualified8Bytes = 7,
+}
+
+const ControlOctetSchema = BitmapSchema({
```

Comment:
I think we should be consistent and always use `Byte` rather than mix terms and use `Octet` sometimes.

Reply:
Agreed with consistency.
I think the specs use Octet everywhere. I personally prefer Byte...
If we use Byte and want to be consistent, this means that we will use ByteString type instead of OctetString type everywhere.

So the choices / tradeoffs are:
1 _ use Byte everywhere but not be consistent with Matter spec naming
2 _  use Octet everywhere but this is less common than Byte
3 _  use Byte in our codebase and only Octet when there is a need to respect Matter specs naming

Probably the best choice is 2, so I should rename Byte to Octet...
WDYT?

Reply:
Done, I renamed Octet to Byte everywhere